### PR TITLE
Add simple script to revert jenkins master

### PR DESCRIPTION
**Pull request by fejta**
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/140)

### CODE REVIEW

File: jenkins/revert-master.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -o errexit
+read -p "Delete master [press enter]"
+gcloud compute instances delete pull-jenkins-master
+read -p "Delete master disks [press enter]"
+gcloud compute disks delete pull-jenkins-master pull-jenkins-master-data pull-jenkins-master-docker
+
+read -p "Recreate master disks from snapshot [press enter]"
+gcloud compute disks create pull-jenkins-master --source-snapshot=cadgdhszak0z --size=500GB
```
Comment (ixdy) [inline jenkins/revert-master.sh]: where do these magic snapshot names come from? can we use something more meaningful?

File: jenkins/revert-master.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -o errexit
+read -p "Delete master [press enter]"
+gcloud compute instances delete pull-jenkins-master
```
Comment (ixdy) [inline jenkins/revert-master.sh]: you're missing implicit --project and --zone on these calls

File: jenkins/revert-master.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+read -p "Delete master disks [press enter]"
+gcloud compute disks delete pull-jenkins-master pull-jenkins-master-data pull-jenkins-master-docker
```
Comment (ixdy) [inline jenkins/revert-master.sh]: maybe we should have some way of verifying there are snapshots first? this is very scary.

File: jenkins/revert-master.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (ixdy) [inline jenkins/revert-master.sh]: this script is also poorly named. it's for PR jenkins only.

File: jenkins/revert-master.sh (hunk 2 above)
Comment (fejta) [inline jenkins/revert-master.sh]: Agreed

File: jenkins/revert-master.sh (hunk 1 above)
Comment (fejta) [inline jenkins/revert-master.sh]: Acknowledged

File: jenkins/revert-master.sh (hunk 4 above)
Comment (fejta) [inline jenkins/revert-master.sh]: Acknowledged

File: jenkins/revert-master.sh (hunk 3 above)
Comment (fejta) [inline jenkins/revert-master.sh]: Sounds like a great improvement if we actually want to invest further in this script. Right now I just want something in source control.

Comment (ixdy) [inline jenkins/revert-master.sh]: perhaps have a large comment that says **DO NOT ACTUALLY USE THIS** then?

Comment (ixdy) [inline jenkins/revert-master.sh]: (it's fine as documentation, but it shouldn't be consider a blessed script)

Comment (fejta) [inline jenkins/revert-master.sh]: I'll be happy to replace this with the script you are creating.

File: jenkins/revert-master.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+set -o errexit
```
Comment (spxtr) [inline jenkins/revert-master.sh]: also `nounset` and `pipefail`

File: jenkins/revert-master.sh (hunk 3 above)
Comment (ixdy) [inline jenkins/revert-master.sh]: what I'm worried about is if Jenkins falls over and someone tries to be helpful and runs this script, not fully understanding what it does, and inadvertently makes things worse.

Comment (spxtr) [inline jenkins/revert-master.sh]: Maybe add an `echo "DO NOT USE THIS SCRIPT" && exit 1` at the top.

**Comment by spxtr** [comment]
What's the status of this?

**Comment by fejta** [comment]
Converted script to python

File: jenkins/master_ctl.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    """Converts us-central1-f into us-central1."""
+    return '-'.join(zone.split('-')[:2])
+
+
+def Snapshot(project, zone, instance):
+    """Snapshot all the disks for this instance."""
+    SnapshotDisks(project, zone, *Disks(instance))
+
+
+def Delete(project, zone, instance):
```
Comment (rmmh) [inline jenkins/master_ctl.py]: This doesn't delete the associated snapshots? Is there any data that might be lost?

Comment (fejta) [inline jenkins/master_ctl.py]: No, it doesn't delete the snapshots.

Any data that on the disks that isn't in a snapshot will be lost.

**Comment by rmmh** [comment]
I'm a little leery of this-- does this mean we might revert Jenkins and forget that we've already had an e2e build #4090? Overwriting previous build results on GCS seems like a very bad result.

Throughout: 4-space indents :-)

**Comment by fejta** [comment]
Do you have an alternative suggestion? I consider this a strong improvement over me manually typing these commands, which is what currently happens.

**Comment by rmmh** [comment]
OK. We can manually override build numbers if necessary -- maybe add a TODO to make this automatic later? http://stackoverflow.com/questions/19645430/changing-jenkins-build-number

**Comment by fejta** [comment]
Added a warning message about this to `Delete()`

**Comment by fejta** [comment]
Also note that thus I mostly just snapshot. Thus far I only had to restore once, and yes it involved some futzing after the fact to get it fully working.

**Comment by rmmh** [comment]
Is there a retention policy / automatic snapshot deletion method?

**Comment by rmmh** [comment]
If this is just packaging up manual steps so they're more broadly available, that's fine too-- just wondering if having too many snapshots is a concern, or if the plan is to take automatic snapshots.

**Comment by fejta** [comment]
There isn't

**Comment by rmmh** [comment]
LGTM % indentation

**Pull request merged**